ulib/experimental/Steel.Reference.fsti

```diff
@@ -151,31 +151,55 @@ val cas_pt (#t:eqtype)
 (** Second version of references: The memory contents are available inside the selector, instead of as an index of the predicate **)
 
 /// An abstract separation logic predicate stating that reference [r] is valid in memory.
-/// (AF) TODO: This should also be indexed by a permission
-val ptr (#a:Type0) (r:ref a) : slprop u#1
+
+val ptrp (#a:Type0) (r:ref a) (p: perm) : slprop u#1
+
+[@@ __steel_reduce__; __reduce__]
+unfold
+let ptr (#a:Type0) (r:ref a) : slprop u#1 = ptrp r full_perm
 
 /// A selector for references, returning the value of type [a] stored in memory
-val ptr_sel (#a:Type0) (r:ref a) : selector a (ptr r)
+
+val ptrp_sel (#a:Type0) (r:ref a) (p: perm) : selector a (ptrp r p)
+
+[@@ __steel_reduce__; __reduce__]
+unfold
+let ptr_sel (#a:Type0) (r:ref a) : selector a (ptr r) = ptrp_sel r _
```

Comment:
Even though it should be the same, I'd make full_perm an explicit argument here, it might cause issues with normalization down the line